geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -365,12 +365,26 @@ public InternalDistributedMember getPreferredNode() {
     }
     getPartitionedRegionStats().incPreferredReadRemote();
 
-    if (locProfiles.length == 1) { // only one choice!
+
+    int index = 0;
+    for (int j = 0; j < locProfiles.length; j++) {
+      Profile tempprofile = locProfiles[j];
+      BucketProfile bp = (BucketProfile) locProfiles[j];
+      if (!bp.isInitializing) {
```

Comment:
Is this block of code just filtering out BucketProfiles that are `isInitializing`? If so, move this logic to a method called something like `int filterOutIntizializing(BucketProfiles[] bucketProfiles)` so that its really clear what is doing on here.

Reply:
Solution updated